Assets/Scripts/Controllers/EnemyController.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class EnemyController : MonoBehaviour
+{
+    [SerializeField] private float speed = 10f;
+
+    private enemyDir dirFacing = enemyDir.Up;
+    private float lookRadius = 4.5f;
+    Transform target;
+    NavMeshAgent agent;
+    private bool moving;
+    private float rayLength = 1.4f;
+    private float rayOffsetX = 0.4f;
+    private float rayOffsetY = 0.4f;
+    private float rayOffsetZ = 0.4f;
+
+
+    public Animator animator;
+    public bool isTurn;
+
+    enum enemyDir
+    {
+        Up,
+        Down,
+        Left,
+        Right
+    }
+
+
+    private Vector3 enemyTargetPosition;
+    private Vector3 enemyStartPosition;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        target = PlayerManager.instance.player.transform; //target.position player Position; transform.position for enemy position
+        animator = GetComponent<Animator>();
+        agent = GetComponent<NavMeshAgent>();
+        agent.updateRotation = false;
+    }
+
+    // Update is called once per frame
+    void Update()
```

Comment:
Since these are duplicates of the methods in PlayerController, let's abstract these to a MovementController class.